### ISSUE TITLE
Fails to build with GCC 14 due to -Wincompatible-pointer-types

### ISSUE THREAD

**Issue by glaubitz** [silx-kit/hdf5plugin#306]
Trying to build hdf5plugin with GCC 14 fails with:

```
[   77s]   INFO:root:gcc -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -DOPENSSL_LOAD_CONF -fwrapv -fno-semantic-interposition -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -IVendor/ -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -g -IVendor/ -O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fPIC -DHAVE_LZ4=1 -DHAVE_ZLIB=1 -DHAVE_ZSTD=1 -DH5_USE_18_API -DHAVE_SNAPPY=1 -DSHUFFLE_SSE2_ENABLED=1 -I/usr/include -Ib'/usr/include\n' -Ib'/usr/include\n' -Ib'/usr/include\n' -Ib'/usr/include\n' -Isrc/hdf5-blosc/src -Ib'/usr/include\n' -I/usr/include/python3.10 -c src/hdf5-blosc/src/blosc_filter.c -o build/temp.linux-x86_64-cpython-310/src/hdf5-blosc/src/blosc_filter.o -std=gnu99 -O3 -ffast-math -pthread -msse2
[   77s]   src/hdf5-blosc/src/blosc_filter.c: In function ‘blosc_filter’:
[   77s]   src/hdf5-blosc/src/blosc_filter.c:183:14: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
[   77s]     183 |     complist = blosc_list_compressors();
[   77s]         |              ^
[   77s]   src/hdf5-blosc/src/blosc_filter.c:184:49: error: passing argument 2 of ‘blosc_compcode_to_compname’ from incompatible pointer type [-Wincompatible-pointer-types]
[   77s]     184 |     code = blosc_compcode_to_compname(compcode, &compname);
[   77s]         |                                                 ^~~~~~~~~
[   77s]         |                                                 |
[   77s]         |                                                 char **
[   77s]   In file included from src/hdf5-blosc/src/blosc_filter.h:8,
[   77s]                    from src/hdf5-blosc/src/blosc_filter.c:19:
[   77s]   /usr/include/blosc.h:357:72: note: expected ‘const char **’ but argument is of type ‘char **’
[   77s]     357 | BLOSC_EXPORT int blosc_compcode_to_compname(int compcode, const char **compname);
[   77s]         |                                                           ~~~~~~~~~~~~~^~~~~~~~
[   77s]   src/hdf5-blosc/src/blosc_filter.c:160:8: warning: unused variable ‘errmsg’ [-Wunused-variable]
[   77s]     160 |   char errmsg[256];
[   77s]         |        ^~~~~~
[   77s]   error: command '/usr/bin/gcc' failed with exit code 1
[   77s]   error: subprocess-exited-with-error
[   77s]   
[   77s]   × Building wheel for hdf5plugin (pyproject.toml) did not run successfully.
[   77s]   │ exit code: 1
[   77s]   ╰─> See above for output.
[   77s]   
[   77s]   note: This error originates from a subprocess, and is likely not a problem with pip.
[   77s]   full command: /usr/bin/python3.10 /usr/lib/python3.10/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py build_wheel /tmp/tmprktchzvt
[   77s]   cwd: /home/abuild/rpmbuild/BUILD/hdf5plugin-4.4.0
[   77s]   Building wheel for hdf5plugin (pyproject.toml): finished with status 'error'
[   77s]   ERROR: Failed building wheel for hdf5plugin
[   77s] Failed to build hdf5plugin
[   77s] ERROR: Failed to build one or more wheels
```

To reproduce, use GCC 14 as the default C/C++ compiler.

**Comment by t20100** [comment]
Thanks for the issue.
I reported it upstream, since this is code we embed from the hdf5-blosc project: https://github.com/Blosc/hdf5-blosc/issues/34

**Comment by glaubitz** [comment]
@t20100 This has already been fixed in https://github.com/Blosc/hdf5-blosc/pull/32/commits/526e3f40420abbdef7e34f4fd01be97eb405bec9. So you just have to pull the latest code from https://github.com/Blosc/hdf5-blosc.

**Comment by t20100** [comment]
I compiled the current tip of this project inside hdf5plugin on ubuntu24.04 with gcc-14 and it compiles fine.

**Comment by t20100** [comment]
This will be fixed by updating: hdf5-blosc to [v1.0.1](https://github.com/Blosc/hdf5-blosc/releases/tag/v1.0.1)
